setupext.py

```diff
@@ -748,12 +694,12 @@ def check(self):
             raise CheckFailed("Mac OS-X only")
         return super().check()
 
-    def get_extension(self):
+    def get_extensions(self):
         sources = [
             'src/_macosx.m'
             ]
         ext = Extension('matplotlib.backends._macosx', sources)
         ext.extra_link_args.extend(['-framework', 'Cocoa'])
         if platform.python_implementation().lower() == 'pypy':
             ext.extra_compile_args.append('-DPYPY=1')
-        return ext
+        yield ext
```

Comment:
Maybe it's clearer to just return a 1-element list here?
```suggestion
        return [ext]
```

Reply:
I think that's actually less consistent with the sole other (non-empty) implementation of get_extensions() (the main list), but can do the change if you insist.

Reply:
While `return` specifies the return value completely, `yield` specifies only one element (of possibly many), so you need more context to infer the full returned result. But I don't really mind.